traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/Order.java

```diff
@@ -83,87 +83,52 @@ public boolean equals(@Nullable Object obj) {
-                        status = new OrderStatus.UnderReview(new PaymentInfo(last4DigitsString));
-                        break;
-                    case "Cancelled":
-                        Assertion.eval(last4DigitsString != null);
-                        status = new OrderStatus.Cancelled(new PaymentInfo(last4DigitsString));
-                        break;
-                    default:
-                        throw new IllegalArgumentException("Unknown status");
-                }
-
-                Assertion.eval(id != null);
-                Assertion.eval(price != null);
-                Assertion.eval(createdDate != null);
-                Assertion.eval(contact != null);
-
-                return new Order(id, price, orderNumber, status, products, createdDate, contact);
+        public Order instantiate(String rawResponse) throws Exception {
+            JSONObjectGLX jsonObject = new JSONObjectGLX(rawResponse);
+
+            String id = jsonObject.getString("id");
+            Price price = new Price.PriceObjectMappingFactory().instantiate(jsonObject.getJSONObject("amount").toString());
+            String orderNumber = jsonObject.getNullableString("referenceNumber");
+            String statusString = jsonObject.getString("status");
+            String last4DigitsString = jsonObject.getNullableString("last4Digits");
+            List<Product> products = new PurchasedProductListMappingFactory().instantiate(jsonObject.getJSONArray("products").toString());
+            Date createdDate = DateHelper.parseISO8601(jsonObject.getString("createdOn"));
+            CustomerContact contact =  new CustomerContact.CustomerContactObjectMappingFactory().instantiate(jsonObject.getString("customer"));
+
+            OrderStatus status = null;
+
+            Assertion.eval(statusString != null);
+
+            switch (statusString) {
+                case "Pending":
+                    status = new OrderStatus.Pending();
+                    break;
+                case "Confirmed":
+                    Assertion.eval(last4DigitsString != null);
+                    status = new OrderStatus.Confirmed(new PaymentInfo(last4DigitsString));
+                    break;
+                case "Declined":
+                    Assertion.eval(last4DigitsString != null);
+                    status = new OrderStatus.Declined(new PaymentInfo(last4DigitsString));
+                    break;
+                case "UnderReview":
+                    Assertion.eval(last4DigitsString != null);
+                    status = new OrderStatus.UnderReview(new PaymentInfo(last4DigitsString));
+                    break;
+                case "Cancelled":
+                    Assertion.eval(last4DigitsString != null);
+                    status = new OrderStatus.Cancelled(new PaymentInfo(last4DigitsString));
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unknown status");
```

Comment:
Same thing here. Backend created a new order status called "Deferred" and its breaking parsing in both platforms. When backend versioning kicks in this will be unnecessary but I am not sure when that is going to happen or how so we have to account for new cases coming in.